src/test/test_circuitpadding.c

```diff
@@ -584,12 +584,12 @@ test_circuitpadding_token_removal_higher(void *arg)
     tt_int_op(mi->histogram[bin_to_remove], OP_EQ, 2);
 
     mi->padding_scheduled_at_usec = current_time - 57;
-    circpad_machine_remove_token(mi);
+    circpad_cell_event_nonpadding_sent((circuit_t*)client_side);
```

Comment:
It's better to use TO_CIRCUIT() for this kind of upcast: it's typesafe, and more compiler-friendly.

Reply:
Heh, not only that, these were all circuit_t * already. I fixed all the instances of this in the circpad tests here: fe6fc3cff894c13120a2fdc507aa09f6a9cc5bb2 (PR #1024)